xrally_kubernetes/tasks/scenarios/volumes/base.py

```diff
@@ -17,22 +17,22 @@
 
 class PodWithVolumeBaseScenario(common_scenario.BaseKubernetesScenario):
     """Base scenario plugin for all pod with volume scenarios."""
-    def run(self, image, name=None, check_cmd=None, command=None,
-            error_regexp=None, volume=None, status_wait=True):
+    def run(self, image, name=None, namespace=None, check_cmd=None,
+            command=None, error_regexp=None, volume=None, status_wait=True):
         """Super class for all kubernetes pod with volume scenarios.
 
         :param image: pod's image
         :param name: pod's name, equals to volume name
+        :param namespace: pod's custom namespace
         :param check_cmd: pod exec command, available if volume_check is True
         :param command: pod container's command
         :param error_regexp: regexp string to search error in pod exec
                response, available if volume_check is True
         :param volume: a dict, which contains `mount_path` and `volume` keys
                with parts of pod's manifest as values
         :param status_wait: wait for pod's status if True
-        :param volume_check: enable check_volume_pod call or not
         """
-        namespace = self.choose_namespace()
+        namespace = namespace or self.choose_namespace()
```

Comment:
Actually, it is not the issue of the current change, but it is better to move `self.choose_namespace()` to the init method, so we will do not need to share used namespace. Rally runner initializes new instances of scenario class for each iteration.

Reply:
Sound reasonable, ok, will do